joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -28,23 +26,52 @@ trait AstCreatorHelper(implicit withSchemaValidation: ValidationMode) { this: As
     val name       = code(node)
     val identifier = identifierNode(node, name, name, Defines.Any)
     val typeRef    = scope.tryResolveTypeReference(name)
-    scope.lookupVariable(name) match {
-      case None if typeRef.isDefined =>
-        Ast(identifier.typeFullName(typeRef.get.name))
-      case None =>
-        val local = localNode(node, name, name, Defines.Any)
-        scope.addToScope(name, local) match {
-          case BlockScope(block) => diffGraph.addEdge(block, local, EdgeTypes.AST)
-          case _                 =>
+
+    node match {
+      case instanceField: InstanceFieldIdentifier =>
+        scope.findFieldInScope(name) match {
+          case None =>
+            val fullName = s"${scope.surroundingTypeFullName.getOrElse(Defines.UnresolvedNamespace)}.$name"
+            scope.pushField(FieldDecl(name, Defines.Any, false, false, node))
+            astForFieldAccess(
+              MemberAccess(
+                DummyNode(identifierNode(instanceField, "this", "this", Defines.Any))(
```

Comment:
Could we refactor `"this"` into a constant? Perhaps under `io.joern.rubysrc2cpg.passes.Defines`